internal/controllers/phase_ensure_addon_instance.go

```diff
@@ -0,0 +1,61 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	addonsv1alpha1 "github.com/openshift/addon-operator/apis/addons/v1alpha1"
+)
+
+// Ensures the presense or absense of an OperatorGroup depending on the Addon install type.
+func (r *AddonReconciler) ensureAddonInstance(
+	ctx context.Context, log logr.Logger, addon *addonsv1alpha1.Addon) (err error) {
+	// not capturing "stop" because it won't ever be reached due to the guard rails of CRD Enum-Validation Markers
+	targetNamespace, _, _, err := r.parseAddonInstallConfig(ctx, log, addon)
+	if err != nil {
+		return err
+	}
+
+	desiredAddonInstance := &addonsv1alpha1.AddonInstance{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      addonsv1alpha1.DefaultAddonInstanceName,
+			Namespace: targetNamespace,
+		},
+		Spec: addonsv1alpha1.AddonInstanceSpec{
+			HeartbeatUpdatePeriod: int64(10 * time.Second),
```

Comment:
TODO: define the default config/spec of `AddonInstance` in a dedicated place. Maybe in `addoninstance_utils.go` or in `addoninstances_types.go`.  Much better than statically defining it like currently in various methods (prone to mistakes).